### ISSUE TITLE
E-mail notifications are sent to members of a Team with no access to issues.

### ISSUE THREAD

**Issue by taai** [go-gitea/gitea#8985]
- Gitea version (or commit ref): 1.11.0+dev-213-g173da1b5a
- Git version: 2.20.1
- Operating system: Debian 10.1
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes, but it is e-mail related, so you will have to reproduce yourself
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

There is an *Oranization* and it has a *Team* with no access to *Issues*. The problem is that the members of this *Team* receive e-mail notifications when a new issues are created. When they open the link they received in e-mail, of course, a 404 page is displayed (weird that it is not 403, but ok), because they don't have access to *Issues* of the *Repository* belonging to *Group*. But they shouldn't have received these notifications if the first place...

## Screenshots

![image](https://user-images.githubusercontent.com/90734/68858202-32f2a500-06ed-11ea-8970-de639776973f.png)

![image](https://user-images.githubusercontent.com/90734/68858518-e6f43000-06ed-11ea-8553-bf3442cb1aad.png)

**Comment by Bryanpedini** [comment]
> a 404 page is displayed (weird that it is not 403, but ok)

That's normal, GitHub also does that, it's to prevent the kind of situations where someone knows there is something there, they want to see what's in there, and they can at least confirm that there there is something, since a 403 is returned.
With a 404, bots, crawlers, random people that got there by accident and most importantly people with bad intentions (mostly security related) get kept off the resource since "for whom has access to the resource, the resource is there, for everyone else the rail 9 and 3/4 doesn't exist. Cit. Harry Potter)

**Comment by lunny** [comment]
Github have no permission control for issues I think. So this should be a bug of gitea.

**Comment by Bryanpedini** [comment]
> Github have no permission control for issues I think.

True, but it has for private repositories, and a 404 is returned instead on GitHub too.
Talking about a 404 instead of a 403 or 401 or whatever, that's normal... "if you have access to it, the resource is there, if you don't, well, you shouldn't know that the resource exists, so I tell you that it does not exists" ;)

**Comment by zeripath** [comment]
Heya sorry this dropped to the bottom of the pile, but could you just confirm this is still an issue?

**Comment by taai** [comment]
@zeripath I just upgraded Gitea to version ```1.12.0+dev-80-g3f1e062d2``` and the issue still exists.

**Comment by zeripath** [comment]
OK basically we need to ensure that people who get sent notifications can actually do things with them.

**Comment by zeripath** [comment]
Yup there is no checking permissions when we send these things. The problem is that the filtering of these IDs is quite expensive and is very difficult to do right now. There's an issue: #9613 that would suggest an easier way of assessing permission for actions.

Now that's not to say that we can't do this.

**Comment by bryanpedini** [comment]
> Yup there is no checking permissions when we send these things.

Bump. (again? glad I have GitHub email notifications for subscribed topics.. :wink: fiuu..)

**Comment by bircni** [comment]
This was fixed in Gitea 1.14.0 by #13820 ("Make sure email recipients can see issue"). Issue/PR notification emails now run each recipient through a repository-unit permission check (`CheckRepoUnitUser`) before sending, so members of a team without access to Issues are excluded and no longer receive these notifications. Closing as resolved; please reopen if you can still reproduce on a current release.
